packages/compiler/test/render3/view/parse_template_options_spec.ts

```diff
@@ -9,22 +9,33 @@
 import {Comment} from '../../../src/ml_parser/ast';
 import {parseTemplate} from '../../../src/render3/view/template';
 
-describe('comments', () => {
+describe('collectCommentNodes', () => {
   it('should include an array of HTML comment nodes on the returned R3 AST', () => {
     const html = `
       <!-- eslint-disable-next-line -->
-      <div *ngFor="let item of items">{{item.name}}</div>
+      <div *ngFor="let item of items">
+        {{item.name}}
+      </div>
+
+      <div>
+        <p>
+          <!-- some nested comment -->
+          <span>Text</span>
+        </p>
+      </div>
     `;
 
     const templateNoCommentsOption = parseTemplate(html, '', {});
     expect(templateNoCommentsOption.comments).toEqual(undefined);
 
-    const templateCommentsOptionDisabled = parseTemplate(html, '', {comments: false});
+    const templateCommentsOptionDisabled = parseTemplate(html, '', {collectCommentNodes: false});
     expect(templateCommentsOptionDisabled.comments).toEqual(undefined);
 
-    const templateCommentsOptionEnabled = parseTemplate(html, '', {comments: true});
-    expect(templateCommentsOptionEnabled.comments!.length).toEqual(1);
+    const templateCommentsOptionEnabled = parseTemplate(html, '', {collectCommentNodes: true});
+    expect(templateCommentsOptionEnabled.comments!.length).toEqual(2);
     expect(templateCommentsOptionEnabled.comments![0] instanceof Comment).toEqual(true);
```

Comment:
There is now a `toBeInstanceOf()` function in Jasmine:
```suggestion
    expect(templateCommentsOptionEnabled.comments![0]).toBeInstanceOf(Comment);
```